### PR TITLE
feat: introduce Qml to parser

### PR DESCRIPTION
**Pull request by itzurabhi**
add the Qml tree sitter language support to all places except test files. 

part of #37

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -135,6 +135,14 @@ add_custom_command(
           ${CMAKE_BINARY_DIR}/compile_commands.json compile_commands.json
   WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/test_data/projects/cpp-project)
 
+  add_custom_command(
```
Comment (itzurabhi) [inline CMakeLists.txt:138]: will remove this, part of work in progress.

File: test_data/projects/CMakeLists.txt [hunk 2]
```diff
@@ -8,3 +8,4 @@
 #
 
 add_subdirectory(cpp-project)
+add_subdirectory(qml-project)
```
Comment (itzurabhi) [inline test_data/projects/CMakeLists.txt:11]: will remove this

File: src/gui/mainwindow.cpp [hunk 3]
```diff
@@ -619,10 +619,11 @@ void MainWindow::updateActions()
     ui->actionExecuteAPI->setEnabled(textDocument != nullptr);
 
     auto *codeDocument = qobject_cast<Core::CodeDocument *>(document);
+    const bool qmlEnabled = codeDocument && qobject_cast<Core::QmlDocument *>(document);
     const bool lspEnabled = codeDocument && codeDocument->hasLspClient();
     ui->actionFollowSymbol->setEnabled(lspEnabled);
     ui->actionSwitchDeclDef->setEnabled(lspEnabled);
-    ui->actionTreeSitterInspector->setEnabled(lspEnabled);
+    ui->actionTreeSitterInspector->setEnabled(lspEnabled || qmlEnabled);
```
Comment (narnaud) [inline src/gui/mainwindow.cpp:626]: checking that it's a codedocument should be enough, as it will have a TreeSitter backend
No need for `qmlEnabled`

File: src/core/qmldocument.cpp [hunk 4]
```diff
@@ -9,12 +9,14 @@
 */
 
 #include "qmldocument.h"
+#include "astnode.h"
```
Comment (narnaud) [inline src/core/qmldocument.cpp:12]: Why this change?

File: src/treesitter/parser.h [hunk 5]
```diff
@@ -38,6 +39,8 @@ class Parser
 
     const TSLanguage *language() const;
 
+    static TSLanguage *getTSLanguageForDocumentType(Core::Document::Type type);
```
Comment (narnaud) [inline src/treesitter/parser.h:42]: as the Parser is using `language` as a name, we could reduce the static name to `getLanguage(Core::Document::Type type)`, no need for TS or re-adding the parameter type.

File: src/treesitter/parser.cpp [hunk 6]
```diff
@@ -62,4 +63,18 @@ const TSLanguage *Parser::language() const
     return ts_parser_language(m_parser);
 }
 
+TSLanguage *Parser::getTSLanguageForDocumentType(Core::Document::Type type)
+{
+    switch (type) {
+    case Core::Document::Type::Qml:
+        return tree_sitter_qmljs();
+    // FIXME: make sane default for other documents.
```
Comment (narnaud) [inline src/treesitter/parser.cpp:71]: In fact, I would put the Q_UNREACHABLE under `default:`, this should be a good default.
Because as it is, it should be impossible to reach that for now.

**Review by narnaud — COMMENTED** [review]
It's getting there, quite happy to see the inspector working with a QML file.

File: src/core/qmldocument.cpp (hunk 4 above)
Comment (itzurabhi) [reply src/core/qmldocument.cpp]: editor doing autoimports :(

File: src/treesitter/parser.h (hunk 5 above)
Comment (itzurabhi) [reply src/treesitter/parser.h]: done

**Review by narnaud — APPROVED** [review]
Thank you!
